app/controllers/transactions_controller.rb

```diff
@@ -5,8 +5,13 @@ class TransactionsController < ApplicationController
 
   def new
     @q = current_organization.transactions.ransack(session[:filter])
-    @transaction = Transaction.new
-    @transfer = Transfer.new
+    if params[:invoice_id]
```

Comment:
@yunixon please make sure here that invoice_id is correct and it's a real Invoice from current_organization. User `before_filter` with set `@invoice` variable. Then use this preparing of @transaction fields only if `@invoice.present?`